quilt/data.py

```diff
@@ -84,8 +84,7 @@ def _keys(self):
         """
         group = self._store.get(self._prefix)
         assert isinstance(group, dict), "{type} {grp}".format(type=type(group), grp=group)
-        assert TYPE_KEY in group
-        group.pop(TYPE_KEY)
+        del group[TYPE_KEY]
```

Comment:
Actually, forgot to ask: is it ok to modify the data here? Does `_store.get` return a new dict each time?